src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -733,7 +769,8 @@ private boolean isNotRelevantSituation(DetailAST ast, int currentType) {
         return starImportOrSlistInsideCaseGroup
                 || colonOfCaseOrDefaultOrForEach
                 || emptyBlockOrType
-                || isArrayInitialization(currentType, parentType);
+                || !allowWhitespaceAroundArrayInit
+                && isArrayInitialization(currentType, parentType);
```

Comment:
We can make this a bit more concise: make a new boolean, something like `isArrayInitializerAndShouldBeChecked` to use in the return statement.

Reply:
I can do this but then `isArrayInitialization` won't be used anywhere. Then it would be almost same like https://github.com/checkstyle/checkstyle/pull/9239#discussion_r571736672

Reply:
done

Reply:
I should have been more clear; as we have done for `emptyBlockOrType`, please make a boolean variable to be used in return statement. I think a method is not necessary in this case.

Reply:
sry, my bad, fixed.

Reply:
Resolved.